chore(flags): add metrics for timeout diagnosis

haacked opened the pull request:
## Problem

We need better insight into how our app is performing.

## Changes

This adds some metrics to help us better understand timeout behavior in the app.

It also adds metrics to help us see whether the connection hoarding fix actually works.

## How did you test this code?

Unit Tests.

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Optional, but helpful for our content team! -->
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->

haacked committed:
chore(flags): add metrics for timeout diagnosis

This adds some metrics to help us better understand timeout behavior in the app.

It also adds metrics to help us see whether the connection hoarding fix actually works.

dmarticus commented on rust/feature-flags/src/metrics/consts.rs:
```diff
@@ -60,3 +60,14 @@ pub const FLAG_EXPERIENCE_CONTINUITY_REQUESTS_COUNTER: &str =
+
+// Timeout tracking and classification
+pub const FLAG_CLIENT_TIMEOUT_COUNTER: &str = "flags_client_timeout_total";
+pub const FLAG_ACQUIRE_TIMEOUT_COUNTER: &str = "flags_acquire_timeout_total";
```
seems like we should use this

haacked replied:
Agreed!

dmarticus reviewed (approved):
can we also get some logs here? I'd love to know more about which queries in particular are taking a long time.

haacked committed:
fix: use FLAG_ACQUIRE_TIMEOUT_COUNTER for pool timeouts

Track pool acquisition timeouts with pool labels to identify which pool is saturated.

haacked committed:
chore(flags): add slow query logging with sql_summary

Adds structured logging for slow queries (>200ms) with `sql_summary` field.

haacked commented:
> can we also get some logs here? I'd love to know more about which queries in particular are taking a long time.

Done.
